etc/compile_check.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+set -e # exit when any command fails
+set -x # show all commands being run
+
+GC=go
+COMPILE_CHECK_DIR="internal/test/compilecheck"
+DEV_MIN_VERSION=1.19
+
+# version will flatten a version string of upto 4 components for inequality
+# comparison.
+function version {
+	echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }';
+}
+
+# compile_check will attempt to build the the internal/test/compilecheck project
+# using the provided Go version. This is to simulate an end-to-end use case.
+# This check will only run on environments where the Go version is greater than
+# or equal to the given version.
+function compile_check {
+	# Change the directory to the compilecheck test directory.
+	cd ${COMPILE_CHECK_DIR}
+
+	MACHINE_VERSION=`${GC} version | { read _ _ v _; echo ${v#go}; }`
+
+	# If the version is not 1.13, then run "go mod tidy"
+	if [ $(version $MACHINE_VERSION) -gt $(version 1.13) ]; then
```

Comment:
This check appears to not work as expected in the Go 1.13 build because the current logic considers 1.13.15 > 1.13. The more correct check is that the Go version is >= 1.15 because that's the oldest minor version with the backported fix.
```suggestion
	if [ $(version $MACHINE_VERSION) -ge $(version 1.15) ]; then
```
P.S. The actual backport versions are [Go 1.15.10](https://github.com/golang/go/issues/44792) and [Go 1.16.2](https://github.com/golang/go/issues/44793), but I don't think we need to be that specific.